### PR TITLE
mds: add a perf counter to record slow replies

### PR DESCRIPTION
**Pull request by Rethan**
Though we have MDS_HEALTH_SLOW_METADATA_IO and MDS_HEALTH_SLOW_REQUEST health alert, but those are not
precise nor accumulated. With slow reply counter compared to reply counter, we can find out the ratio
of slow requests through perf dump.

Signed-off-by: haoyixing <haoyixing@kuaishou.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by Rethan** [comment]
jenkins test make check

File: src/mds/MDSRank.cc [hunk 1]
```diff
@@ -3350,6 +3350,8 @@ void MDSRank::create_logger()
                             "exi", PerfCountersBuilder::PRIO_INTERESTING);
     mds_plb.add_u64_counter(l_mds_imported_inodes, "imported_inodes", "Imported inodes",
                             "imi", PerfCountersBuilder::PRIO_INTERESTING);
+    mds_plb.add_u64_counter(l_mds_slow_reply, "slow_reply", "Slow replies", "slr",
```
Comment (vshankar) [inline src/mds/MDSRank.cc:3353]: `slow reply' is a bit confusing - how about using `slow requests'?

**Comment by vshankar** [comment]
@Rethan we probably want to backport this - please create a tracker for the same.

File: src/mds/MDSRank.cc (hunk 1 above)
Comment (Rethan) [reply src/mds/MDSRank.cc:3353]: IMO  l_mds_request and l_mds_reply are not necessarily equal, so named it as l_md_slow_reply which is more related to l_mds_reply. l_mds_slow_request and l_mds_slow_reply both works for me. If you prefer l_mds_slow_request I will change to it.

Comment (vshankar) [reply src/mds/MDSRank.cc:3353]: Ah ok - let's keep it as `l_mds_slow_reply` then. thx!

**Comment by Rethan** [comment]
> e probably want to backport this - please create a tracker for the same.
Create a tracker: https://tracker.ceph.com/issues/55126, not sure it's style is correct or not ;-)

**Comment by Rethan** [comment]
> @Rethan we probably want to backport this - please create a tracker for the same.

Create a tracker: https://tracker.ceph.com/issues/55126, not sure it's style is correct or not ;-)

**Comment by neesingh-rh** [comment]
> > @Rethan we probably want to backport this - please create a tracker for the same.
> 
> Create a tracker: https://tracker.ceph.com/issues/55126, not sure it's style is correct or not ;-)

@Rethan Pls add this to the desc. of commit also:) eg:_Fix: https://tracker.ceph.com/issues/55126_

**Comment by Rethan** [comment]
> > > @Rethan we probably want to backport this - please create a tracker for the same.
> > 
> > 
> > Create a tracker: https://tracker.ceph.com/issues/55126, not sure it's style is correct or not ;-)
> 
> @Rethan Pls add this to the desc. of commit also:) eg:_Fix: https://tracker.ceph.com/issues/55126_

updated

**Comment by neesingh-rh** [comment]
> > > > @Rethan we probably want to backport this - please create a tracker for the same.
> > > 
> > > 
> > > Create a tracker: https://tracker.ceph.com/issues/55126, not sure it's style is correct or not ;-)
> > 
> > 
> > @Rethan Pls add this to the desc. of commit also:) eg:_Fix: https://tracker.ceph.com/issues/55126_
> 
> updated

Pls follow the pull request template:
- Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

**Comment by Rethan** [comment]
jenkins test windows

**Comment by Rethan** [comment]
jenkins test make check

**Comment by Rethan** [comment]
jenkins test make check

**Comment by Rethan** [comment]
jenkins test make check

**Comment by vshankar** [comment]
> jenkins test make check

please rebase and update. (sometime it does not trigger the checks)

**Review by vshankar — APPROVED** [review]

**Pull request merged**